eng/build-job.yml

```diff
@@ -197,11 +150,15 @@ jobs:
         condition: always()
 
     # Publish product output directory for consumption by tests.
-    - task: PublishBuildArtifacts@1
-      displayName: Publish product build
-      inputs:
-        pathtoPublish: $(Build.SourcesDirectory)/bin/Product/$(osGroup).$(archType).$(_BuildConfig)
-        artifactName: ${{ format('BinDir_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
+    - template: /eng/upload-artifact-step.yml
+      parameters:
+        rootFolder: '$(productBuildRootFolderPath)'
+        includeRootFolder: false
+        archiveFile: $(Build.StagingDirectory)/$(productBuildArtifactName)$(archiveExtension)
+        archiveType: $(archiveType)
+        tarCompression: $(tarCompression)
+        artifactName: $(productBuildArtifactName)
+        displayName: 'product build'
```

Comment:
Update `displayName`.

Reply:
Well, could you please clarify how exactly you'd like the name to be changed? The template ends up creating two steps, one zipping up the subtree and the other publishing it to Azure, and it calls them 'Zip ' + displayName / 'Publish ' + displayName. So in this case we'll end up with exactly the same name ('Publish product build') for this particular step.

Reply:
I guess the problem may be caused by the fact that my choice of the parameter name "displayName" for the upload/download-test-artifact template was somewhat unfortunate as it overlaps the "standard" meaning of displayName in job steps. If you think it can clarify things, I can rename the parameter to something like 'artifactDisplayName' to emphasize this fact.

Reply:
Sorry. I was under the impression this should be whatever the name of the comment is above the template. For example this one would be `Publish product output directory` or perhaps a distillation of the template name.

Reply:
You can see the names in the current CI run:

https://dev.azure.com/dnceng/public/_build/results?buildId=369018

Could you please briefly skim over the jobs in the pipeline and let me know if the naming seems reasonable to you or propose any changes as you see fit? After all, I don't deny I'm not a native English speaker so I can easily imagine some of my naming may be cumbersome or outright incorrect.

Thanks

Tomas

Reply:
These all look good. I guess I don't understand where they are being used, but the names a clear. Thank you!

Reply:
Well, this part of the change mostly stemmed from Jarret's comment that zipping up the artifacts and publishing them to Azure as a single file is about 10 times faster than UploadBuildArtifacts. While this might be somewhat too optimistic, it indeed is faster so I modified the various pipelines to treat the product build (bin/product/os.arch.cfg) like this. I introduced the templates for doing this (zipping up a subtree and pushing it to Azure as an artifact and the counterpart download & unzip) in my first infra change splitting test build and run jobs so I just reused them here.